kopf/_cogs/configs/diffbase.py

```
@@ -123,6 +125,18 @@ def build(
             extra_fields: Iterable[dicts.FieldSpec] | None = None,
     ) -> bodies.BodyEssence:
         essence = super().build(body=body, extra_fields=extra_fields)
+
+        # Remove extra fields if specified
+        for path in self._remove_fields:
+            d = essence
+            keys = path.split('.')
+            for k in keys[:-1]:
+                # If current parent field is not a dict then the path is invalid; skip removal
+                if not isinstance(d.get(k, {}), dict):
+                    break
+                d = d.get(k, {})
+            d.pop(keys[-1], None)
```

I have a strong suspicion that this code semantically duplicates `dicts.remove()` from `kopf._cogs.structs.dicts`, which has an advantage of also removing the empty parents.

In that case, you can avoid writing numerous edge-case tests for the fields removal, as the removal is tested. Only a single test that the removal is happening/attempted at all would suffice.

---

`dicts.remove` produces an error `TypeError: 'str' object does not support item deletion` when the user tries to delete a field that is actually a string (or any type that doesn't support field deletion). For example trying to delete the field `a.b` will produce this error because `a` is a string and has no subfields:
```json
{
  "a": "b"
}
```
In my code I handle it with the check `if not isinstance(d.get(k, {}), dict):` and just skip it. also `d = d.get(k, {})` ensures that non existent fields are also skipped silently.
Do you think I should still catch these errors and handle them silently or let the errors propagate?

ps: thanks for the review!

---

I ended up silently ignoring non-existent fields, and fields that don't support item deletion.

---

note: `dicts.remove` already ignores non-existent fields, so catching `TypeError` exception is just for ignoring fields that don't support item deletion.